modules/EvseManager/Charger.cpp

```diff
@@ -993,11 +1001,32 @@ bool Charger::cancel_transaction(const types::evse_manager::StopTransactionReque
             shared_context.current_state = EvseState::ChargingPausedEVSE;
         }
 
+        // FIXME(evgeny): We disable pwm here since we need to get signed meter values.
+        // Maybe state machine should be refactored and cancelTransaction should trigger
+        // transition to EvseState::Finished, with active transaction.
+        // This way, the signed meter values will be retrieved there.
+        pwm_off();
```

Comment:
I agree we should change the state machine flow here, also to make basic charging more aligned with ISO charging. But let's maybe do this in a separate PR, otherwise we cannot get this into the March release anymore as this requires a bit of thinking and testing to make sure we don't break anything else.

So to keep changes minimal for the moment I think it is enough to move the pwm_off() into the else line 1001. Then it would only stop PWM on basic charging and not trigger an emergency shutdown on ISO. On ISO we have sent the ISO message to the car that we would like to stop, but it of course takes a bit of time for the car to actually stop drawing power etc. The same is true for basic charging, worst case the car can draw power for 6s still in C1 until we force power off under load (IEC61851-1). I would accept that in this case though, the worst case means we give 6s away for free on non-cooperative cars. ISO shutdown should also be within few seconds.

With that in mind, the question is if PWM really needs to be switched off here at all. It will be switched off very quickly after the power meter reading anyway once we are in state ChargingPausedEVSE for basic charging. It does not hurt though as long as we only do it for basic charging and not for ISO.

Reply:
Moved.